src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzIamStep.java

```diff
@@ -50,6 +52,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
           SnapshotDuosFlightUtils.getFirecloudGroup(context);
       derivedPolicies.addReadersItem(duosFirecloudGroup.getFirecloudGroupEmail());
     }
+    if (snapshotRequestModel.getContents().get(0).getMode()
+        == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
+      var snapshotFirecloudGroupEmail =
+          workingMap.get(SnapshotWorkingMapKeys.SNAPSHOT_FIRECLOUD_GROUP_EMAIL, String.class);
+      if (snapshotFirecloudGroupEmail == null) {
```

Comment:
If we need to do a null check, I think we should do it before setting the value in the flight map (in `CreateSnapshotSamGroupStep`). If we get here and this value is missing or empty, it should be a NPE like any other time the map doesn't contain a value we expect to be there.

Reply:
After discussing, I've removed the null check.